### DISCUSSION TITLE
GPUMD interface

### DISCUSSION THREAD

**Discussion by Youhaojen** [mir-group/pair_nequip#40] [Ideas]
Dear NequIP Developers,

I would like to express my appreciation for the excellent performance of NequIP on Lammps. It has proven to be a valuable tool in our work. However, I was curious if there is a potential opportunity to extend its capabilities by creating an interface for GPUMD. Given the exceptional quality and functionality of NequIP, I believe that the addition of a GPUMD interface would further amplify its value and expand its range of applications.

Best regards,
Hao-Jen You

**Comment by Linux-cpp-lisp** [comment]
Hi Hao-Jen,

Thanks for your enthusiasm about NequIP!

At a high level, yes, you can certainty write an interface for NequIP/Allegro models based on `pair_nequip`/`pair_allegro` into other codes, including those in Fortran (see CP2K).  You can see in the Kokkos implementations how to get pointers to blobs of GPU CUDA memory into PyTorch, and how to extract the results back into plain CUDA memory. You'd need to adapt all the preprocessing logic to the formats in your own code.

Certainly please let us all know if you make any development in this direction!
